Sourcery/Parsing/FileParser.swift

```diff
@@ -452,6 +455,160 @@ extension FileParser {
+                        .trimmingPrefix("{")
+                        .trimmingSuffix("}")
+                        .trimmingCharacters(in: .whitespacesAndNewlines)
+                    return (returnTypeName, body)
+                } else {
+                    // actual type name should be set when last parameter is processed
+                    return ("", "")
+                }
+            } else { return nil }
+        } else { return nil }
+    }
+
+    private func parseSubscriptAccessLevel(_ keyPrefix: inout String, definedIn: Type?) -> (readAccessLevel: AccessLevel, writeAccessLevel: AccessLevel, isFinal: Bool) {
+        var readAccessLevel: AccessLevel = definedIn.flatMap({ AccessLevel(rawValue: $0.accessLevel) }) ?? .`internal`
+        var writeAccessLevel: AccessLevel = .none
+        var isFinal: Bool = false
+
+        let accessLevels: [AccessLevel] = [.`private`, .`fileprivate`, .`internal`, .`public`, .`open`]
+        var readAllPrefixes: Bool = false
+
+        while !readAllPrefixes {
+            var readReadAccessLevel: Bool = false
+            var readWriteAccessLevel: Bool = false
+            var readFinalAttribute: Bool = false
+
+            if let _readAccessLevel = accessLevels.first(where: { keyPrefix.trimSuffix($0.rawValue) }) {
+                readAccessLevel = _readAccessLevel
+                keyPrefix = keyPrefix.trimmingCharacters(in: .whitespacesAndNewlines)
+                readReadAccessLevel = true
+            }
+            if let _writeAccessLevel = accessLevels.first(where: { keyPrefix.trimSuffix("\($0.rawValue)(set)") }) {
+                writeAccessLevel = _writeAccessLevel
+                keyPrefix = keyPrefix.trimmingCharacters(in: .whitespacesAndNewlines)
+                readWriteAccessLevel = true
+            }
+            if keyPrefix.trimSuffix("final") {
+                isFinal = true
+                keyPrefix = keyPrefix.trimmingCharacters(in: .whitespacesAndNewlines)
+                readFinalAttribute = true
+            }
+            readAllPrefixes =
+                (readReadAccessLevel && readWriteAccessLevel && readFinalAttribute) ||
+                (!readReadAccessLevel && !readWriteAccessLevel && !readFinalAttribute)
+        }
+
+        return (readAccessLevel, writeAccessLevel, isFinal)
+    }
+
+    private func lastNonCommentStartingLine(_ keyPrefix: String) -> Line? {
+        let keyPrefixLines = keyPrefix.lines()
+        for keyPrefixLine in keyPrefixLines.reversed() {
+            let line = keyPrefixLine.content.trimmingCharacters(in: .whitespacesAndNewlines)
+            let isCommmentLine = line.hasPrefix("//") || line.hasPrefix("/*")
+            if !isCommmentLine && !line.isEmpty {
+                return keyPrefixLine
+            }
+        }
+        return nil
+    }
+
+    internal func parseSubscript(_ source: [String: SourceKitRepresentable], definedIn: Type? = nil) -> Subscript? {
```

Comment:
this isn't something you need to change but just a general recommendation (that I think I first read in Clean Coder) to place the private methods below the public methods, so that when you read the code the higher abstraction is read first, that way when you read code first for `parseSubscript()` and usually you don't need to know the implementation detail of `lastNonCommentStartingLine` to understand the intent. So higher abstraction first, and implementation details later will make code more readable in the long run